devito/types.py

```diff
@@ -348,6 +348,7 @@ def __new__(cls, *args, **kwargs):
             newobj._indices = indices
             newobj._shape = cls.__shape_setup__(**kwargs)
             newobj._dtype = cls.__dtype_setup__(**kwargs)
+            newobj._staggered = cls.__staggered_setup__(newobj, **kwargs)
```

Comment:
don't pass ``newobj``, just have ``__staggered_setup__`` return what you want to store in ``_staggered``

Reply:
Necessary, needs to access `obj.indices` in the function

Reply:
ah, so don't make it a classmethod. For example, look at __halo_setup__